gems/linked_in/lib/linked_in/connection.rb

```diff
@@ -21,43 +21,69 @@
 
 module LinkedIn
   class Connection
+
+    def get_request(path, access_token)
+      config = self.class.config
+
+      http = Net::HTTP.new("api.linkedin.com", 443)
+      http.use_ssl = true
+      #http.verify_mode = OpenSSL::SSL::VERIFY_NONE
+      http
+
+      request = Net::HTTP::Get.new(path)
+      request['Authorization'] = "Bearer #{access_token}"
+      response = http.request(request)
+
+      response
+    end
+
     def get_service_user_info(access_token)
-      body = access_token.get('/v1/people/~:(id,first-name,last-name,public-profile-url,picture-url)').body
+      body = get_request('/v1/people/~:(id,first-name,last-name,public-profile-url,picture-url)', access_token).body
       data = Nokogiri::XML(body)
       service_user_id = data.css("id")[0].content
       service_user_name = data.css("first-name")[0].content + " " + data.css("last-name")[0].content
       service_user_url = data.css("public-profile-url")[0].content
       return service_user_id, service_user_name, service_user_url
     end
 
-    def get_access_token(token, secret, oauth_verifier)
-      consumer = self.class.consumer
-      request_token = OAuth::RequestToken.new(consumer, token, secret)
-      request_token.get_access_token(:oauth_verifier => oauth_verifier)
+    def get_service_user_data_export(access_token)
+      # #########
+      # TODO: here is where we call into the LInkedIn API. Change this to download all of their data
+      # #########
+      Rails.logger.debug("### get_service_user_data_export - begin")
+      return true
     end
 
-    def request_token(oauth_callback)
-      consumer = self.class.consumer
-      consumer.get_request_token(:oauth_callback => oauth_callback)
+    def authorize_url(return_to, nonce)
+      config = self.class.config
+      "https://www.linkedin.com/oauth/v2/authorization?response_type=code&client_id=#{config['api_key']}&state=#{nonce}&redirect_uri=#{CGI.escape(return_to)}"
```

Comment:
Instead of hardcoding this path, should we use the authorize_url specified in the client_options field of app/models/account_authorization_config/linked_in.rb ?